tests/structured_conf/data/dataclasses.py

```diff
@@ -475,3 +475,15 @@ class MissingStructuredConfigField:
 class ListClass:
     list: List[int] = field(default_factory=lambda: [])
     tuple: Tuple[int, int] = field(default_factory=lambda: (1, 2))
+
+
+@dataclass
+class GenericList:
+    list: List = field(default_factory=lambda: [])  # type: ignore
```

Comment:
put something in the list, like [1,2,3] and verify the content in your test as well.
same for the dict.